### PR TITLE
feat(tracing): make tracer._sampler public

### PR DESCRIPTION
**Pull request by ZStriker19**
In anticipation of lazy sampling, users may create egress points that we cannot account for, and therefore miss sampling the span on. For example, an integration that where a user is manually injecting trace headers into the outgoing request will miss calling tracer.sampler.sample() which could lead to broken traces. To avoid that we'll make sure to advise users when to call sampler in the lazy sampling PR: https://github.com/DataDog/dd-trace-py/pull/8308

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by ZStriker19** [comment]
I did think of that but didn't try the implementation. I'll give it a go to investigate if it's better.

File: releasenotes/notes/sampler_public-0570e2ce18bc604b.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    tracing: Makes ``tracer.sampler`` public, now named ``tracer.sampler``. This should not be called
```
Comment (ZStriker19) [inline releasenotes/notes/sampler_public-0570e2ce18bc604b.yaml:4]: this needs to be updated if we decide to pull the sampler out of the tracer.

**Comment by emmettbutler** [comment]
@ZStriker19 I think that for this change to remain small enough to merge, the Tracer should continue to hold a Sampler instance. My suggestion about removing that attribute was meant more as a long-term possibility than a suggestion for right now.

The point I wanted and still want to get across is that exposing sampling functionality via `tracer.sampler` rather than `tracer.do_sampling` limits our ability to change the underlying implementation in the future. Of course, deprecations are cheap, so this isn't a dealbreaker. Instead it's just a suggestion about a readily available way to futureproof the new public API surface you're considering here.

**Review by emmettbutler — APPROVED** [review]
I think an interface to the specific functionality you want to make public would be more futureproof than having the interface be unrestricted access to the `.sampler` attribute as you've written here. With that said, if you want to go ahead with this approach I'm ok with it.

**Pull request merged**